File: scripts/scrapers/skills.py
from .Scraper import Scraper

class SkillsScraper(Scraper):
    def scrape(self) -> list[dict]:
        self._load_page_details("/details/skills/", scroll_count=5)
        return [
            s.inner_text()
            for s in self.page.locator("main section a span[aria-hidden]").all()
            if "endorse" not in s.inner_text().lower()
            and "experiences" not in s.inner_text().lower()
        ]
        

File: scripts/browser-scrapers/skills.js
/**
 * LinkedIn Skills Scraper - Run in browser console on:
 * linkedin.com/in/YOUR_USERNAME/details/skills/
 *
 * Scroll down first to load lazy content (skills load on scroll).
 * Copies JSON result to clipboard. Also returns the data.
 */
(function scrapeSkills() {
  const skills = [...new Set(
    [...document.querySelectorAll("main section a span[aria-hidden]")]
      .map((s) => s.innerText.trim())
      .filter((text) => !text.toLowerCase().includes("endorse") && !text.toLowerCase().includes("experiences"))
  )];

  const result = JSON.stringify(skills, null, 2);
  navigator.clipboard.writeText(result);
  console.log("Skills data copied to clipboard. Found", skills.length, "skills.");
  return skills;
})();
